### ISSUE TITLE
Error when building: "use of undeclared identifier 'aligned_alloc'"

### ISSUE THREAD

**Issue by rullinoiz** [projectM-visualizer/projectm#378]
I ran `./autogen.sh` and `./configure --enable-sdl` and I got this error when building:

```            
make  all-recursive
make[1]: Entering directory '/Users/OllinRuiz/projectm'
Making all in src
make[2]: Entering directory '/Users/OllinRuiz/projectm/src'
Making all in libprojectM
make[3]: Entering directory '/Users/OllinRuiz/projectm/src/libprojectM'
Making all in Renderer
make[4]: Entering directory '/Users/OllinRuiz/projectm/src/libprojectM/Renderer'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/Users/OllinRuiz/projectm/src/libprojectM/Renderer'
Making all in NativePresetFactory
make[4]: Entering directory '/Users/OllinRuiz/projectm/src/libprojectM/NativePresetFactory'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/Users/OllinRuiz/projectm/src/libprojectM/NativePresetFactory'
Making all in MilkdropPresetFactory
make[4]: Entering directory '/Users/OllinRuiz/projectm/src/libprojectM/MilkdropPresetFactory'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/Users/OllinRuiz/projectm/src/libprojectM/MilkdropPresetFactory'
make[4]: Entering directory '/Users/OllinRuiz/projectm/src/libprojectM'
depbase=`echo wipemalloc.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../..  -Wall -Wchar-subscripts -Wformat-security -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\""/usr/local/share/projectM\"" -I../../vendor -include ../../config.h -DSYSCONFDIR=\""/usr/local/etc"\" -I../../src/libprojectM -I../../src/libprojectM/Renderer -I../../vendor   -g -O2 -pthread -stdlib=libc++ -std=c++11 -MT wipemalloc.lo -MD -MP -MF $depbase.Tpo -c -o wipemalloc.lo wipemalloc.cpp &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../.. -Wall -Wchar-subscripts -Wformat-security -Wpointer-arith -Wshadow -Wsign-compare -Wtype-limits -DDATADIR_PATH=\"/usr/local/share/projectM\" -I../../vendor -include ../../config.h -DSYSCONFDIR=\"/usr/local/etc\" -I../../src/libprojectM -I../../src/libprojectM/Renderer -I../../vendor -g -O2 -pthread -stdlib=libc++ -std=c++11 -MT wipemalloc.lo -MD -MP -MF .deps/wipemalloc.Tpo -c wipemalloc.cpp  -fno-common -DPIC -o .libs/wipemalloc.o
wipemalloc.cpp:60:11: error: use of undeclared identifier 'aligned_alloc'
    mem = aligned_alloc( align, size );
          ^
1 error generated.
make[4]: *** [Makefile:614: wipemalloc.lo] Error 1
make[4]: Leaving directory '/Users/OllinRuiz/projectm/src/libprojectM'
make[3]: *** [Makefile:655: all-recursive] Error 1
make[3]: Leaving directory '/Users/OllinRuiz/projectm/src/libprojectM'
make[2]: *** [Makefile:457: all-recursive] Error 1
make[2]: Leaving directory '/Users/OllinRuiz/projectm/src'
make[1]: *** [Makefile:527: all-recursive] Error 1
make[1]: Leaving directory '/Users/OllinRuiz/projectm'
make: *** [Makefile:417: all] Error 2
```

I searched around the internet for this error and they said it was defined in a version of C. I already upgraded all my packages and I have all the dependencies installed, so outdated compilers can't be an issue.

Edit: forgot to mention I'm on macOS 10.15.5

**Comment by revmischa** [comment]
Yeah I get this unless I build with xcode. Unfortunately I don't know what the solution is other than use xcode on macOS.

**Comment by rullinoiz** [comment]
> Yeah I get this unless I build with xcode. Unfortunately I don't know what the solution is other than use xcode on macOS.

guess i gotta build with xcode. thanks for the info

**Comment by rullinoiz** [comment]
seems to have worked. thanks

**Comment by revmischa** [comment]
It's still an issue that would be nice to fix, something is wrong with autoconf on macOS I think

**Comment by milkdropper** [comment]
I tried to get this to build without xcode because I'm still having problems with static linking in macOS.

From my research in macOS you need to use posix_memalign instead of aligned_alloc. But if you look at the source of wipemalloc.cpp you will see this is already handled based on the definition of HAVE_ALIGNED_ALLOC. If you manually set this to 0 it will compile without xcode. Alternatively we could change the logic from checking HAVE_ALIGNED_ALLOC to instead checking if it's __APPLE__.

Sharingt his info in case it helps others.

**Comment by milkdropper** [comment]
Further info-sharing: once you bypass the above error it will compile fine. But you won't have text menus at all (because USE_TEXT_MENU is not defined). If you define this you will then encounter compilation errors due to missing OpenGL headers in the renderer. I added this to Renderer.cpp:

#include <OpenGL/gl.h>
#include <OpenGL/glu.h>
#include <OpenGL/glext.h>
#include <GLUT/glut.h>

After that all will compile normally without xcode... but my issue with static linking is not gone. It won't run on a third-party PC that doesn't have SDL2 installed. Even though it was a dead end for me I hope this helps someone else :) If anyone has a tip please share.

**Comment by revmischa** [comment]
This issue should be fixed in #384. @milkdropper please add a config option in `configure.ac` to detect whatever library (glut?) gltext requires and enable `USE_TEXT_MENU` if it's found
